### PR TITLE
[WIP]Wildcard imports

### PR DESCRIPTION
**Pull request by claudiu-coman**
Hi @trotterdylan! I read the conversation in #192 and I thought I'd take a shot at wildcard imports. The code should be functional, I've updated the tests too.
If the idea is the right one, the next step for me would be to start trimming that code down. I can isolate some code in reusable methods and remove some of the duplication.

Before I take this PR further, can you tell me if I'm heading in the right direction?

### CODE REVIEW

File: compiler/stmt_test.py [hunk 1]
```diff
@@ -366,14 +366,25 @@ def testImportFromFutureParseError(self):
...
+        from re import *
+        print compile"""))
+    self.assertEqual(0, result[0])
+    self.assertIn('<function compile at',result[1])
+
+    self.assertEqual((0, 'False\n'), _GrumpRun(textwrap.dedent("""\
+        from re import *
+        print 'sre_compile' in globals()""")))
+
+    # 'time' does not define __all__
```
Comment (trotterdylan) [inline compiler/stmt_test.py]: This seems like an implementation detail that could change. Maybe we need to create a couple dummy test modules for the purposes of testing this functionality.

File: compiler/stmt.py [hunk 2]
```diff
@@ -748,6 +925,19 @@ def _import_native(self, name, values):
...
 
+  def _import_wildcard(self, node):
+    """If __all__ is defined, attempts to bind everything from it,
+    falls back on __dict__ otherwise.
+
+    Note: items in __dict__ whose names start with '_' are skipped.
+    """
+    module_name = node.module
+
+    with self.block.alloc_temp() as members, \
```
Comment (trotterdylan) [inline compiler/stmt.py]: IIUC members is a temporary that will be filled up, used to populate the globals dict and then thrown away. I would just populate the globals dict as I iterate over the imported module's members and avoid the temporary.

**Comment by claudiu-coman** [comment]
It struck me that I could write the code in Go, as I got more familiar with the codebase. It was a bit too late and I was closer to finishing the Python code. And I was afraid Go would be the better solution. :) I'll start working on that on Monday, excited about how this will turn out.

BTW, the Travis jobs for my commits seem to take a lot of time. Job #679 seems to have been running for over 50 mins now. Is that normal?

**Comment by trotterdylan** [comment]
Sounds good.

The OS X job on Travis is slow. Usually it suffices to validate that the
Linux job was successful. It's usually pretty quick.

**Pull request closed without merging**